In [2]:
from scipy import stats

## Probability Distribution Practice

### Lt Col Horton

In this notebook, I will provide some practice problems that will help reinforce your understanding of probability distributions. Throughout this practice, I highly encourage you to refer to the `scipy` documentation online. While you are there, feel free to explore areas we don't cover in this notebook (particularly plotting and randomization). 

For each of the tasks below **_1)_** define a random variable that will help you answer the question; **_2)_** state the distribution and parameters of that random variable; **_3)_** determine the expected value and variance of that random variable. 

I will demonstrate using **_1.1_** below. 

#### Problem 1

The T-6 training aircraft is used during UPT. Suppose that on each training sortie, aircraft return with a maintenance-related failure at a rate of 1 per 100 sorties. 

**_1.1_** Find the probability of no maintenance failures in 15 sorties. 

**_1.2_** Find the probability of at least two maintenance failures in 15 sorties. 

**_1.3_** Find the probability of at least 30 successful (no mx failures) sorties before the first failure.

**_1.4_** Find the probability of at least 50 successful sorties before the third failure. 



##### Demonstration using 1.1

Find the probability of no maintenance failures in 15 sorties.

$X$: the number of maintenance failures in 15 sorties. 

$X\sim \textsf{Bin}(n=15, p=0.01)$

$E(X) = np = 15*0.01 = 0.15$

$V(X) = np(1-p) = 15*0.01*0.99 = 0.1485$

Probability of no maintenance failures, $P(X=0)$:  

In [5]:
stats.binom.pmf(0,15,0.01)

0.8600583546412884

The probability of getting no maintenance failures in 15 sorties is about 0.86. It is worth taking a moment to make sure this value makes sense. Since failures are fairly unlikely, and the **expected** number of failures (0.15) is close to 0, then 0 failures should be a fairly likely outcome and a probability of 0.86 makes sense. 

#### Problem 2

On a given Saturday, suppose vehicles arrive at the USAFA North Gate according to a Poisson process at a rate of 40 arrivals per hour. 

**_2.1_** Find the probability no vehicles arrive in 10 minutes. 

**_2.2_** Find the probability at least 50 vehicles arrive in an hour. 

**_2.3_** Find the probability that at least 5 minutes will pass before the next arrival.

**_2.4_** Find the probability that the next vehicle will arrive between 2 and 10 minutes from now. 

**_2.5_** Find the probability that at least 7 minutes will pass before the next arrival, given that 2 minute have already passed. Compare this answer to **_2.3_**. This is an example of the *memoryless* property of the exponential distribution.

**_2.6_** Fill in the blank. There is a probability of 90% that the next vehicle will arrive within __ minutes. This value is known as the 90% percentile of the random variable. 


#### Problem 3

Suppose there are 12 male and 7 female cadets in a classroom. I select 5 completely at random (without replacement). 

**_3.1_** Find the probability I select no female cadets. 

**_3.2_** Find the probability I select more than 2 female cadets. 

#### Problem 4

Suppose PFT scores in the cadet wing follow a normal distribution with mean 330 and standard deviation 50. 

**_4.1_** Find the probability a randomly selected cadet has a PFT score higher than 450. 

**_4.2_** Find the probability a randomly selected cadet has a PFT score within 2 standard deviations of the mean.

**_4.3_** Find $a$ and $b$ such that 90% of PFT scores will be between $a$ and $b$. 

**_4.4_** Find the probability a randomly selected cadet has a PFT score higher than 450 given he/she is among the top 10% of cadets. 

#### Problem 5

Suppose time until computer errors on the F-35 follows a Gamma distribution with mean 20 hours and variance 10.  

**_5.1_** Find the probability that 50 hours pass without a computer error. 

**_5.2_** Find the probability that 75 hours pass without a computer error, given that 25 hours have already passed. Dose the memoryless property apply to the Gamma distribution? 

**_5.3_** Find $a$ and $b$: There is a 95% probability time until next computer error will be between $a$ and $b$.  